staging/src/k8s.io/kubectl/pkg/cmd/debug/profiles.go

```diff
@@ -21,39 +21,246 @@ import (
+		UseHostNamespaces(pod)
+
+	case podCopy:
+		RemoveLabelsAndProbes(pod)
+		AllowProcessTracing(pod, containerName, podutils.Containers)
+		ShareProcessNamespace(pod)
+
+	case ephemeral:
+		AllowProcessTracing(pod, containerName, podutils.EphemeralContainers)
+	}
+
+	return nil
+}
+
+func (p *baselineProfile) Apply(pod *corev1.Pod, containerName string, target runtime.Object) error {
+	style, err := getDebugStyle(pod, target)
+	if err != nil {
+		return fmt.Errorf("baseline profile: %s", err)
+	}
+
+	ClearSecurityContext(pod, containerName, podutils.Containers|podutils.EphemeralContainers)
+
+	switch style {
+	case podCopy:
+		RemoveLabelsAndProbes(pod)
+		ShareProcessNamespace(pod)
+
+	case ephemeral, node:
+		// no additional modifications needed
 	}
+
+	return nil
+}
+
+func (p *restrictedProfile) Apply(pod *corev1.Pod, containerName string, target runtime.Object) error {
+	style, err := getDebugStyle(pod, target)
+	if err != nil {
+		return fmt.Errorf("restricted profile: %s", err)
+	}
+
+	DisallowRoot(pod, containerName, podutils.Containers|podutils.EphemeralContainers)
+	DropCapabilities(pod, containerName, podutils.Containers|podutils.EphemeralContainers)
+
+	switch style {
+	case node:
+		ClearSecurityContext(pod, containerName, podutils.Containers)
+
+	case podCopy:
+		ShareProcessNamespace(pod)
+
+	case ephemeral:
+		// no additional modifications needed
+	}
+
+	return nil
+}
+
+// RemoveLabelsAndProbes removes labels from the pod and remove probes
+// from all containers of the pod.
+func RemoveLabelsAndProbes(p *corev1.Pod) {
+	p.Labels = nil
+	for i := range p.Spec.Containers {
+		p.Spec.Containers[i].LivenessProbe = nil
+		p.Spec.Containers[i].ReadinessProbe = nil
+	}
+}
+
+// MountRootPartition mounts the host's root path at "/host" in the container.
+func MountRootPartition(p *corev1.Pod, containerName string) {
+	const volumeName = "host-root"
+	p.Spec.Volumes = append(p.Spec.Volumes, corev1.Volume{
+		Name: volumeName,
+		VolumeSource: corev1.VolumeSource{
+			HostPath: &corev1.HostPathVolumeSource{Path: "/"},
+		},
+	})
+	podutils.VisitContainers(&p.Spec, podutils.Containers, func(c *corev1.Container, _ podutils.ContainerType) bool {
+		if c.Name != containerName {
+			return true
+		}
+		c.VolumeMounts = append(c.VolumeMounts, corev1.VolumeMount{
+			MountPath: "/host",
+			Name:      volumeName,
+		})
+		return false
+	})
+}
+
+// UseHostNamespaces configures the pod to use the host's network, PID, and IPC
+// namespaces.
+func UseHostNamespaces(p *corev1.Pod) {
+	p.Spec.HostNetwork = true
+	p.Spec.HostPID = true
+	p.Spec.HostIPC = true
+}
+
+// ShareProcessNamespace configures all containers in the pod to share the
+// process namespace.
+func ShareProcessNamespace(p *corev1.Pod) {
+	p.Spec.ShareProcessNamespace = pointer.BoolPtr(true)
+}
+
+// ClearSecurityContext clears the security context for the container.
+func ClearSecurityContext(p *corev1.Pod, containerName string, mask podutils.ContainerType) {
```

Comment:
nit: mask is unnecessary. Since we're searching by unique name it can always be `Containers|EphemeralContainers` (or even `AllContainers`: I'd prioritize readability over optimization for this infrequently used loop)

Same for all of the other helpers that take a mask.

Reply:
Ah, I didn't know that we could unambiguously identify by the name alone, simplified in d3583dcd60bc6d5ec2968c2e9852334672ba5852